tests/policy_engine/cue/test_cue_validator.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module tests the CUE policy validator."""
+
+import json
+import os
+from pathlib import Path
+
+import pytest
+
+from macaron.policy_engine.cue.cue_policy import CUEPolicy
+from macaron.policy_engine.cue.cue_validator import get_target, validate_policy
+
+RESOURCES_PATH = Path(__file__).parent.joinpath("resources")
+
+
+@pytest.mark.parametrize(
+    "policy_path",
+    [
+        os.path.join(RESOURCES_PATH, "invalid_policies", "1.cue"),
+        os.path.join(RESOURCES_PATH, "invalid_policies", "2.cue"),
+        os.path.join(RESOURCES_PATH, "invalid_policies", "no_file.cue"),
+    ],
+)
+def test_make_policy(policy_path: str) -> None:
+    """Test making policies from invalid CUE policy files.
+
+    A CUE policy that misses "target" field is considered invalid.
+    """
+    assert not CUEPolicy.make_policy(policy_path=policy_path)
+
+
+@pytest.mark.parametrize(
+    ("policy_path", "expected"),
+    [
+        (os.path.join(RESOURCES_PATH, "valid_policies", "1.cue"), "urllib3/urllib3"),
+        (os.path.join(RESOURCES_PATH, "valid_policies", "2.cue"), ""),
+    ],
+)
+def test_get_target(policy_path: str, expected: str) -> None:
+    """Test getting target from valid CUE policies."""
+    policy = CUEPolicy.make_policy(policy_path=policy_path)
```

Comment:
> If there is a bug in `CUEPolicy.make_policy` and it returns None for these policy files even thought it's supposed to return a `CUEPolicy` instance, these test cases would be ignored. This comment also applies for `test_validate_policy`.

Yes, and that's on purpose. The invalid tests are tested separately in `test_make_policy` and the assumption for `test_get_target` and `test_validate_policy` is that the policies are valid (see the docstring).

Reply:
I think even we assume that the input policies are valid, if there is an issue with `CUEPolicy.make_policy` that return None in those cases (which is not expected), this test will fail to capture that (because the if inside branch is not run). I think it's best if we `assert policy` to make sure that we cover all logic branch of the test.

Reply:
See commit [545550f](https://github.com/oracle-samples/macaron/pull/90/commits/545550fb90df046bda5066f346a78001b506ea7b).